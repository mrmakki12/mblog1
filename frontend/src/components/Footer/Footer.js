import React from "react";
// styles 
import './styles/footer.css';

export const Footer = () => {
    return (
        <footer data-testid='footer' className="footer">
            <p>Made with <img className="footer-img" src='/images/heart.png' alt='heart'/> By: <a target='_blank' rel='noreferrer' href="https://tyreeckcodes.com">Tyreeck Makki!</a></p>
        </footer>
    )
}