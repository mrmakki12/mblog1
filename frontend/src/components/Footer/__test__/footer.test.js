// test setup 
import React from "react";
// component to test
import { Footer } from '../Footer.js';
// testing 
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);

// test 1 render without crashing 
test('render Footer without crashing', () => {
    render(<Footer />);
});

// test 2 render with proper text 
test('make sure it has the text', () => {
    const { getByText } = render(<Footer />);

    expect(getByText(/made with/i)).toBeDefined();
    expect(getByText(/by:/i)).toBeDefined();
    expect(getByText(/tyreeck makki!/i)).toBeDefined();
});

// test 3 test if has anchor 
test('make sure it has an anchor to my portfolio', () => {
    render(<Footer />);

    expect(screen.getByRole('link')).toHaveAttribute('href', 'https://tyreeckcodes.com');
});

// snapshot
test("matches snapshot", () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
});