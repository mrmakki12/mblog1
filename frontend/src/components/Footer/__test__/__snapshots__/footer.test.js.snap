// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<footer
  className="footer"
  data-testid="footer"
>
  <p>
    Made with 
    <img
      alt="heart"
      className="footer-img"
      src="/images/heart.png"
    />
     By: 
    <a
      href="https://tyreeckcodes.com"
      rel="noreferrer"
      target="_blank"
    >
      Tyreeck Makki!
    </a>
  </p>
</footer>
`;
